src/main/java/uk/gov/hmcts/reform/idam/health/vault/KeyVaultClientProvider.java

```diff
@@ -4,6 +4,6 @@
 
 public interface KeyVaultClientProvider {
 
-    KeyVaultClient getClient(String clientId, String clientKey);
+    KeyVaultClient getClient(String clientId, String clientKey, String msiUrl);
```

Comment:
Maybe we can create a `KeyVaultConfig` class to encapsulate those values (clientId, clientKey, msiUrl, keyVaultBaseUri)?
We can pass it the `environment` and it can extract whatever properties it needs and initialise itself.